### PR TITLE
Update index.md

### PR DESCRIPTION
**Pull request by Rick-Anderson**
[Review URL](https://review.docs.microsoft.com/en-us/aspnet/core/mvc/razor-pages/?branch=pr-en-us-3594)
@tdykstra  can you think of a better way to break up the RO sentence?

### CODE REVIEW

File: aspnetcore/mvc/razor-pages/index.md [hunk 1]
```diff
@@ -132,13 +133,18 @@ The *MyApp/Pages/Contact.cshtml* file:
...
 
 The basic flow of `OnPostAsync` is:
 
 1. Check for validation errors.
-1. If there are no errors, save the data and redirect -
-1. Else, show the page again with the validation errors.
+1. If there are no errors, save the data and redirect.
+1. Else if errors, show the page again with the validation message.
 
 When the data is entered successfully, the `OnPostAsync` handler method calls the `RedirectToPage` helper method to return an instance of `RedirectToPageResult`. This is a new action result similar to `RedirectToAction` or `RedirectToRoute` but customized for pages. In the preceding sample it redirects back to the same URL as the current page (`/Contact`). Later I'll show how to redirect to a different page.
```
Comment (scottaddie) [inline aspnetcore/mvc/razor-pages/index.md]: I recommend a comma after "In the preceding sample".

File: aspnetcore/mvc/razor-pages/index.md [hunk 2]
```diff
@@ -256,7 +263,7 @@ Add a *_ViewImports.cshtml* file:
 @addTagHelper "*, Microsoft.AspNetCore.Mvc.TagHelpers"
```
Comment (scottaddie) [inline aspnetcore/mvc/razor-pages/index.md]: Dan's preference is to remove the double quotes from the `@addTagHelper` directive.

File: aspnetcore/mvc/razor-pages/index.md [hunk 3]
```diff
@@ -373,7 +380,7 @@ This example adds a confirmation message and redirects back to the home page:
 </div>
 ```
 
-We've  added another page (*MyApp/Pages/Index.cshtml*), and are redirecting to it using `RedirectToPage("/Index")`. The string `/Index` is the name of the page we just added, and can be used with `Url.Page(...)`, `<a asp-page="..." />` or `RedirectToPage`.
+We've added another page (*MyApp/Pages/Index.cshtml*), and are redirecting to it using `RedirectToPage("/Index")`. The string `/Index` is the name of the page we just added, and can be used with `Url.Page(...)`, `<a asp-page="..." />` or `RedirectToPage`.
 
 The page name is just the path to the page from the root *MyApp/Pages* folder (including a leading `/`). It seems simple, but this is much more feature rich than just hardcoding a URL. This is URL generation using [routing](xref:mvc/controllers/routing), and can generate and encode parameters according to how the route is defined in the destination path.
```
Comment (scottaddie) [inline aspnetcore/mvc/razor-pages/index.md]: Should the word "it" be added after ", and"?

Comment (tdykstra) [inline aspnetcore/mvc/razor-pages/index.md]: antecedent for "it" wouldn't be clear -- I would split the sentence up, and clarify the last clause:  how is a route "defined in the destination path"?

File: aspnetcore/mvc/razor-pages/index.md [hunk 4]
```diff
@@ -132,13 +133,18 @@ The *MyApp/Pages/Contact.cshtml* file:
...
+
+The `Async` naming suffix is optional but is often used by convention. The code that's in `OnPostAsync` in the preceding example looks similar to what you would normally write in a controller. This is typical for pages. Most of the MVC primitives like model binding, validation, and action results are shared.
 
 The basic flow of `OnPostAsync` is:
 
 1. Check for validation errors.
-1. If there are no errors, save the data and redirect -
-1. Else, show the page again with the validation errors.
+1. If there are no errors, save the data and redirect.
+1. Else if errors, show the page again with the validation message.
```
Comment (tdykstra) [inline aspnetcore/mvc/razor-pages/index.md]: If you have "If errors" do you need "else"?  Also, might be better for consistency to make 2 and 3 parallel: either "if errors/no errors" or "if there are errors/no errors"

File: aspnetcore/mvc/razor-pages/index.md [hunk 5]
```diff
@@ -148,7 +154,7 @@ The `Contact` property is using the new `[BindProperty]` attribute to opt-in to
 
 Rather than using `@model` here, we're taking advantage of a special new feature for pages. By default, the generated `Page`-derived class *is* the model. This means that features like model binding, tag helpers, and HTML helpers all *just work* with the properties defined in `@functions`. Using a *view model* with Razor views is a best practice. With pages, you get a view model automatically. 
 
-Notice that this Page also uses `@inject` for dependency injection, which is the same as traditional Razor views - this generates the `Db` property that is used in `OnPostAsync`. Injected (`@inject`) properties are set before handler methods run.
+Notice that this Page also uses `@inject` for dependency injection, which is the same as traditional Razor views. The `@inject` statement generates the `Db` property that is used in `OnPostAsync`. Injected (`@inject`) properties are set before handler methods run.
```
Comment (tdykstra) [inline aspnetcore/mvc/razor-pages/index.md]: Is "also" needed here?

File: aspnetcore/mvc/razor-pages/index.md [hunk 6]
```diff
@@ -242,9 +249,9 @@ Add a layout page for the HTML skeleton, and set the `Layout` property from `_Vi
 @{ Layout = "_Layout"; }
 ```
 
-Note that we  placed the layout in the *MyApp/Pages* folder. Pages look for other views (layouts, templates, partials) hierarchically, starting in the same folder as the current page. This means that a layout in the *MyApp/Pages* folder can be used from any page.
+Note that we placed the layout in the *MyApp/Pages* folder. Pages look for other views (layouts, templates, partials) hierarchically, starting in the same folder as the current page. This means that a layout in the *MyApp/Pages* folder can be used from any Razor Page.
```
Comment (tdykstra) [inline aspnetcore/mvc/razor-pages/index.md]: Wouldn't Razor page in reference to a specific page be lowercase page? Razor Pages as a feature is uppercase Page.

File: aspnetcore/mvc/razor-pages/index.md (hunk 5 above)
Comment (scottaddie) [inline aspnetcore/mvc/razor-pages/index.md]: I think "also" is a banned word.

File: aspnetcore/mvc/razor-pages/index.md (hunk 3 above)
Comment (Rick-Anderson) [inline aspnetcore/mvc/razor-pages/index.md]: >The string `/Index` is the name of the page 

That's not the name of the page. The page name is *Index.cshtml* - That's part of the URI to access the page.

**Comment by Rick-Anderson** [comment]
@scottaddie  ready to merge?

File: aspnetcore/mvc/razor-pages/index.md [hunk 7]
```diff
@@ -373,9 +381,15 @@ This example adds a confirmation message and redirects back to the home page:
...
+We've added another page (*MyApp/Pages/Index.cshtml*), and are redirecting to it using `RedirectToPage("/Index")`. The string `/Index` is part of the URI to access the the preceding page. The string `/Index` can be used to generate URIs to this page. For example:
+
+
+* `Url.Page("/Index", ...)`
+* `<a asp-page="/Index">My Index Page</a>`
+* `RedirectToPage("/Index")`
+
 
-The page name is just the path to the page from the root *MyApp/Pages* folder (including a leading `/`). It seems simple, but this is much more feature rich than just hardcoding a URL. This is URL generation using [routing](xref:mvc/controllers/routing), and can generate and encode parameters according to how the route is defined in the destination path.
+The page name is the path to the page from the root *MyApp/Pages* folder (including a leading `/`). It seems simple, but this is much more feature rich than just hardcoding a URL. This is URL generation using [routing](xref:mvc/controllers/routing), and can generate and encode parameters according to how the route is defined in the destination path.
```
Comment (scottaddie) [inline aspnetcore/mvc/razor-pages/index.md]: Recommend a dash between "feature" and "rich".

File: aspnetcore/mvc/razor-pages/index.md [hunk 8]
```diff
@@ -132,23 +133,29 @@ The *MyApp/Pages/Contact.cshtml* file:
...
-1. Check for validation errors.
-1. If there are no errors, save the data and redirect -
-1. Else, show the page again with the validation errors.
+Check for validation errors.
 
-When the data is entered successfully, the `OnPostAsync` handler method calls the `RedirectToPage` helper method to return an instance of `RedirectToPageResult`. This is a new action result similar to `RedirectToAction` or `RedirectToRoute` but customized for pages. In the preceding sample it redirects back to the same URL as the current page (`/Contact`). Later I'll show how to redirect to a different page.
+*  If there are no errors, save the data and redirect.
+*  If there are errors, show the page again with the validation message.
+
+When the data is entered successfully, the `OnPostAsync` handler method calls the `RedirectToPage` helper method to return an instance of `RedirectToPageResult`. This is a new action result similar to `RedirectToAction` or `RedirectToRoute` but customized for pages. In the preceding sample, it redirects back to the same URL as the current page (`/Contact`). Later I'll show how to redirect to a different page.
```
Comment (scottaddie) [inline aspnetcore/mvc/razor-pages/index.md]: Recommend a comma after "Later".

**Pull request merged**